e2e/cypress/integration/team_settings/teams_spec.js

```diff
@@ -211,4 +220,175 @@ describe('Teams Suite', () => {
+
+        // Save new team description
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+
+        // Team tooltip should show description
+        verifyToolTip('#headerTeamName', 'This is the best team');
+    });
+
+    it('MM-T2318 Allow anyone to join this team', () => {
+        // # UI
+
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click();
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team description menu item
+        cy.findByText('Allow any user with an account on this server to join this team').should('be.visible').click();
+
+        // # Change team description in the input
+        cy.get('#teamOpenInvite').click();
+
+        // Save new team description
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+
+        // # Functionality
+
+        cy.visit('/login');
+        cy.apiLogin(newUser);
+
+        // * Verify if the user is redirected to the Select Team page
+        cy.url().should('include', '/select_team');
+        cy.get('#teamsYouCanJoinContent').should('be.visible');
+
+        // # Select test team name
+        cy.findByText(testTeam.display_name, {timeout: TIMEOUTS.HALF_MIN}).should('be.visible').click();
+        ignoreUncaughtException();
+
+        // # Verify Town square is visible
+        cy.url().should('include', `/${testTeam.name}/channels/town-square`);
+        cy.findByText('Beginning of Town Square').should('be.visible');
+    });
+
+    it('MM-T2322 Do not allow anyone to join this team: UI', () => {
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click();
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team description menu item
+        cy.findByText('Allow any user with an account on this server to join this team').should('be.visible').click();
+
+        // # Change team description in the input
+        cy.get('#teamOpenInviteNo').click();
+
+        // Save new team description
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+    });
+
+    it('MM-T2322 Do not allow anyone to join this team: Functionality', () => {
+        cy.apiLogin(newUser);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        // # Open the hamburger menu
+        cy.get('#teamIconInitial').should('be.visible').click();
```

Comment:
Got failing test here too. It failed 'cause `newUser` is created but not member of any team.
Also, please make `MM-T2322` in exactly one `it` block only so test result can be mapped and saved into TM4J.